src/main/java/com/spotify/confidence/Confidence.java

```diff
@@ -0,0 +1,136 @@
+package com.spotify.confidence;
+
+import com.google.common.annotations.Beta;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Maps;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.spotify.confidence.shaded.flags.resolver.v1.ResolveFlagsResponse;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+@Beta
+public class Confidence implements EventSender, Contextual {
+
+  private Map<String, ConfidenceValue> context = Maps.newHashMap();
+
+  private final Contextual parent;
+
+  private final EventSenderEngine eventSenderEngine;
+  private final FlagResolverClient flagResolverClient;
+
+  Confidence(
+      @Nullable Contextual parent,
+      EventSenderEngine eventSenderEngine,
+      FlagResolverClient flagResolverClient) {
+    this.parent = parent;
+    this.eventSenderEngine = eventSenderEngine;
+    this.flagResolverClient = flagResolverClient;
+  }
+
+  @Override
+  public ConfidenceValue.Struct getContext() {
+    final ConfidenceValue.Struct parentContext =
+        parent != null ? parent.getContext() : ConfidenceValue.Struct.EMPTY;
+
+    final Map<String, ConfidenceValue> mergedContext = Maps.newHashMap();
+
+    for (Map.Entry<String, ConfidenceValue> entry : parentContext.asMap().entrySet()) {
+      if (!context.containsKey(entry.getKey())) {
+        mergedContext.put(entry.getKey(), entry.getValue());
+      }
+    }
```

Comment:
Extract to a private function returning a stream of context entries. That can then be filtered.

Reply:
Fixed as part of [0c60c59](https://github.com/spotify/confidence-openfeature-provider-java/pull/88/commits/0c60c59e21136f8f5d0f442458fef4d49afdc329)